### PR TITLE
Add import onnx model section to the book

### PR DESCRIPTION
**Pull request by antimora**
### Related Issues/PRs

https://github.com/burn-rs/burn/issues/200

### Changes

Added a section on importing ONNX model.

### Testing

Generated a book and reviewed if rendered correctly.

### CODE REVIEW

**Review by Luni-4 — APPROVED** [review]
It's fine for me from a semantic perspective! Thanks a lot! 

Perhaps we could add some `onnx` examples of real networks to better understand what it is and a link to the specification, but we can do that in later PRs

File: burn-book/src/import/onnx-model.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+If you've been working with other deep learning frameworks like PyTorch, it's likely that you've
+exported model weights before. PyTorch, for instance, lets you save model weights using its
+`torch.save()` function. Yet, to port this model to another framework, you face the arduous task of
+manually recreating the architecture in the destination framework before loading in the weights. Not
+only is this method tedious, but it's also error-prone and hinders smooth interoperability between
+frameworks.
+
+## Enter ONNX: The One Format to Rule Them All
+
+ONNX (Open Neural Network Exchange) is designed to solve such complications. It's an open-standard
```
Comment (louisfd) [inline burn-book/src/import/onnx-model.md:21]: Maybe provide an external link?

File: burn-book/src/import/onnx-model.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+   fine-tuned using Burn's native training loop.
+
+3. **Portability**: As the model is converted to Rust source code, it can be compiled into
+   WebAssembly for browser execution. Likewise, this approach is beneficial for no-std embedded
+   devices.
+
+4. **Optimization**: Rust's compiler can further optimize the generated code for target
+   architectures, thereby improving performance.
+
+As far as we know, Burn is the only deep learning framework that employs this technique, making it
```
Comment (louisfd) [inline burn-book/src/import/onnx-model.md:63]: Unclear which technique you are referring to

**Review by louisfd — APPROVED** [review]
Great! I like how it is clearly separated in steps. 
I have only minor comments, approving in advance!

File: burn-book/src/import/onnx-model.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+is precisely the scenario where importing models into Burn can be a game-changer.
+
+## Traditional Methods: The Drawbacks
+
+If you've been working with other deep learning frameworks like PyTorch, it's likely that you've
+exported model weights before. PyTorch, for instance, lets you save model weights using its
+`torch.save()` function. Yet, to port this model to another framework, you face the arduous task of
+manually recreating the architecture in the destination framework before loading in the weights. Not
+only is this method tedious, but it's also error-prone and hinders smooth interoperability between
+frameworks.
```
Comment (nathanielsimard) [inline burn-book/src/import/onnx-model.md:17]: Sometimes it's necessary and that's fine. When using framework specific features, it's just not a possibility to serialize into a standard; the standard is always behind new innovations. So I would just reframe a bit and enphasis on _standard_ models.

File: burn-book/src/import/onnx-model.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+## Enter ONNX: The One Format to Rule Them All
```
Comment (nathanielsimard) [inline burn-book/src/import/onnx-model.md:19]: I would remove `The One Format to Rule Them All` 😅

File: burn-book/src/import/onnx-model.md (hunk 2 above)
Comment (nathanielsimard) [reply burn-book/src/import/onnx-model.md:63]: Yeah, I think a sentence is necessary at the beginning of the paragraph on code generation.

**Review by nathanielsimard — CHANGES_REQUESTED** [review]

**Review by nathanielsimard — APPROVED** [review]

**Pull request merged**